modules/swagger-parser-v3/src/main/java/io/swagger/v3/parser/util/RefUtils.java

```diff
@@ -136,27 +149,38 @@ public static String readExternalRef(String file, RefFormat refFormat, List<Auth
             throw new RuntimeException("Ref is not external");
         }
 
-        String result;
+        String result = null;
 
         try {
             if (refFormat == RefFormat.URL) {
-
                 result = RemoteUrl.urlToString(file, auths);
-
             } else {
                 //its assumed to be a relative file ref
                 final Path pathToUse = parentDirectory.resolve(file).normalize();
```

Comment:
Should this parentDirectory be the directory of where the $ref exists or the directory of the main file which is read first?

I am asking this because this seems to be the behavior of swagger-ui. And to make my swagger spec work with both I had to use links.

Reply:
is the directory where the file is